### DISCUSSION TITLE
tagged_vlans getting re-created every time

### DISCUSSION THREAD

**Discussion by EpiqSty** [netbox-community/ansible_modules#674] [Q&A] [answered]
Hi, before raising issue i just wanted to ask if it's expected by design:

```
changed: [cs2] => (item=device: cs2; 
name: ae1.0; 
mode:Tagged; 
tagged_vlans:[162, 518, 526];
untagged_vlan:__omit_place_holder__5ed565359be399c8be3a9b5207d1914af5e9f3f3;)
--- before
+++ after
@@ -1,7 +1,7 @@
 {
     "tagged_vlans": [
         162,
-        526,
-        518
+        518,
+        526
     ]
 }
```
As you can see i'm sending list of tagged vlans (using `netbox_device_interface`), sorted by their keys in Netbox - but Ansible module still showing changes, `tagged_vlans` are getting re-created on every run.
Is there any way to avoid such useless changes?

```
$ pip freeze
ansible==4.9.0
ansible-core==2.11.7
Jinja2==2.11.1
pynetbox==6.2.0
```
```
ansible_collections
Collection    Version
------------- -------
netbox.netbox 3.1.2  
```

Thanks in advance!

**Comment by EpiqSty** [comment, marked as answer]
OK, i've found the solution, Netbox expects that list will be sorted by Vlan ID , not key or name.

**Reply by rodvand** [reply]
Sounds a bit like we should do that sort in the module, allowing the VID to be input in any order. If you feel like we should have a look at it, you can create an issue.

**Reply by EpiqSty** [reply]
@rodvand thanks! For now, i have implemented workaround already, so for my case it's not really urgent, but sorting within the module would make lot of sense just because:

- it's some Netbox-internal specifics 
- which is not reflected in the docs
- and diff output adds confusion, as it mentioning Netbox VLAN_ID keys as identifiers, but at fact sorting is happening by VLAN IDs.

So, perhaps for now this information can be at least searched here, in case if it's critical for someone else it can be found and turned to an issue.
